header-rewrite-filter/header_rewrite.cc

```diff
@@ -84,12 +84,21 @@ HttpHeaderRewriteFilter::HttpHeaderRewriteFilter(HttpHeaderRewriteFilterConfigSh
             return;
           }
           // make sure this boolean variable doesn't already exist in the map
-          if (set_bool_processors_->find(boolName) != set_bool_processors_->end()) {
-            fail("redefinition of boolean variable");
-            setError();
-            return;
+          if (isRequest) {
```

Comment:
Nit: this could reduce nesting by changing it to
```
if (isRequest && request_set_bool_processors_->find(boolName) != request_set_bool_processors_->end()) {
⋮ 
} else if(!isRequest && response_set_bool_processors_->find(boolName) != response_set_bool_processors_->end()) {
```

It's a minuscule performance cost, for the gain of reducing Cyclomatic Complexity.

The extra potential bool check is a tiny cost, no additional branches, won't ever cause a mispredict, pipeline flush, etc.

I wouldn't object to either, but I'd personally prefer the nesting/Cyclomatic Complexity reduction, especially since this is at config-load-time not http-request-time.